### ISSUE TITLE
About the re-implement performance

### ISSUE THREAD

**Issue by zaynmi** [yalesong/pvse#21]
I try to reproduce your model on coco using the command:
```
python3 train.py --data_name coco --cnn_type resnet152 --wemb_type glove --margin 0.1 --max_violation --num_embeds 2 --img_attention --txt_attention --mmd_weight 0.01 --div_weight 0.1 --batch_size 256
```
with the pytorch 1.1.0, torchvision 0.3.0 on single RTX2080Ti.

But the loss stays at 0.2001 since epoch 4, and the final performance is bad.
The evaluation result of your provided checkpoint is OK, so I don't think it's the problem with the environment.
Is there any suggestion on this problem?
Any help would be appreciated. Thank you!

**Comment by yalesong** [comment]
Hi there! Not really sure what might be causing your issue. It's been a while since I worked on this so my memories are fading. I'd suggest trying with a simpler model (e.g., disable MIL by setting `--num_embeds 1`, disable attention components, etc.), check the convergence/performance, and slowly add the options back into the model while monitoring the progress.

**Comment by yalesong** [comment]
Closing due to inactivity

**Comment by kdwonn** [comment]
I have also experienced the same issue but I think I found the solution. Turning on the `--img-finetune` and `--txt-finetune` will fix it. :)

**Comment by sanghoooon** [comment]
For prospective users: 

Equivantly, you can just replace `self.legacy` argument in `EncoderText` to `self.training` that is already built in `nn.Module` for simplicity: change `if self.legacy:` to `if self.training:`

I have successfully reproduced the results in the main paper. 
Many thanks to insight from @kdwonn
